### ISSUE TITLE
[fix] Step indicator defines `aria-label` without `role`

### ISSUE THREAD

**Issue by adunkman** [trussworks/react-uswds#3271]
**ReactUSWDS Version & USWDS Version:** 
- trussworks/react-uswds@9.1.0 
- uswds/uswds@3.13.0

**Describe the bug**

The step indicator fails automated accessibility test ([aria-prohibited-attr](https://dequeuniversity.com/rules/axe/4.8/aria-prohibited-attr)) because the `div` cannot use `aria-label` without a valid `role`.

**Expected behavior**

Step indicator passes automated accessibility scans.

**Additional context**

- The [USWDS component](https://designsystem.digital.gov/components/step-indicator/) does not define an `aria-label`.

**Comment by adunkman** [comment]
One potential fix would be to specify `role="progressbar"` but I'm not sure of the implications of this on screen readers. I would generally defer to USWDS on this, which would be to remove `aria-label` instead.
